UTRAnnotator.pm

```diff
@@ -0,0 +1,1286 @@
+          $uSTOP_lost_AltStopDistanceToCDS = "NA";
+        }
+        if (($length-$start_pos) % 3){
+          $uSTOP_lost_FrameWithCDS = "outOfFrame";
+        }
+        else {
+          $uSTOP_lost_FrameWithCDS = "inFrame";
+        }
+        #find evidence from sorf
+
+        $uSTOP_lost_evidence= (exists $self->{uORF_evidence})? $self->find_uorf_evidence($UTR_info,$chr,$start_pos): "NA";
+
+        my %uORF_effect = (
+          "uSTOP_lost_AltStop" => $uSTOP_lost_AltStop,
+          "uSTOP_lost_AltStopDistanceToCDS" => $uSTOP_lost_AltStopDistanceToCDS,
+          "uSTOP_lost_FrameWithCDS" => $uSTOP_lost_FrameWithCDS,
+          "uSTOP_lost_KozakContext" => $uSTOP_lost_KozakContext,
+          "uSTOP_lost_KozakStrength" => $uSTOP_lost_KozakStrength,
+          "uSTOP_lost_Evidence" => $uSTOP_lost_evidence,
+        );
+
+        $output_flag = $output_flag? $output_flag."&"."5_prime_UTR_stop_codon_loss_variant":"5_prime_UTR_stop_codon_loss_variant";
+        $output_effects = $output_effects? $output_effects."&".$self->transform_hash_to_string(\%uORF_effect):$self->transform_hash_to_string(\%uORF_effect);
+      }
+    }
+  }
+
+  $result{'uSTOP_lost_flag'} = $output_flag;
+  $result{'uSTOP_lost_effect'} = $output_effects;
+
+  return \%result;
+}
+
+sub uAUG_lost {
+  # Description: annotate if a five_prime_UTR_varint removes a start codon of an existing uORF
+
+  my ($self, $variant_info, $UTR_info, $mut_pos, $mut_utr_seq, $existing_ref_uORF, $start) = @_;
+
+  my $chr = $variant_info->{chr};
+  my $pos = $variant_info->{pos};
+  my $ref = $variant_info->{ref};
+  my $alt = $variant_info->{alt};
+
+  my @sorted_starts = @{$UTR_info->{start}};
+  my @sequence = split //, $UTR_info->{seq};
+  my $utr_length = @sequence;
+  my $strand = $UTR_info->{strand};
+
+  #return annotators
+  my $uAUG_lost_type = "";   # the uORF type with the reference allele - uORF, inframe_oORF, outOfFrame_oORF
+  my $uAUG_lost_DistanceToCDS = ""; # the distance of the lost uAUG to CDS
+  my $uAUG_lost_DistanceToSTOP = ""; # the distance of the lost uAUG to stop codon (could be in CDS)
+  my $uAUG_lost_KozakContext = ""; # the Kozak context sequence of the lost uAUG
+  my $uAUG_lost_KozakStrength = ""; # the strength of KozakContext of the lost uAUG
+  my $uAUG_lost_evidence = ""; # whether this uAUG has any evidence of being translated
+
+  my $current_kozak = "";
+  my $current_kozak_strength = "";
+
+  #indicate whether the variant ever disrupts a stop codon
+  my $output_flag = "";
+  #indicate whether the variant ever disrupts the uORF that being analyzed
+  my $flag_uORF=0;
+
+  #the relative position of input variant in the UTR sequence
+
+  my %result = (); #result is a hash table with two elements: $flag and $output_effects
+  my $output_effects="";
+
+  my $ref_coding = $self->{ref_coding};
+  my $alt_coding = $self->{alt_coding};
+
+  my @mut_utr_seq = split //,$mut_utr_seq;
+
+  # my @start = @{$self->get_ATG_pos(\@sequence)};
```

Comment:
I forgot to remove this comment, you can remote it:
`# my @start = @{$self->get_ATG_pos(\@sequence)};`